src/core/Val.ml

```diff
@@ -1907,7 +1907,7 @@ struct
           let body =
             lazy begin
               let func = Value.act phi func in
-              apply func el
+              apply func @@ Value.act phi el
```

Comment:
@favonia This seemed like a bug, but not sure. what do you think?

Reply:
Indeed! However, is there a reason not to use the helper functions in Face.ml?

Reply:
The main reason is I didn't understand how to use those functions.

Reply:
Let me update it